src/app/campo/campo.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component } from '@angular/core';
+import { DatosService } from '../datos.service'; /* esto escrito a mano */
+
+@Component({
+  selector: 'app-campo',
+  templateUrl: './campo.component.html',
+  styleUrls: ['./campo.component.css']
+})
+export class CampoComponent {
+ 
+  pokemones: Array<Record<string, string | number>> = [];
+
+  constructor(public DatosService: DatosService) { /* lo de los paréntes escrito a mano */
+
+    var anteriorX: Array<number> = [];
+    var anteriorY: Array<number> = [];
+
+    if (DatosService.fase == "campo") {
+      for (let i = 0; i <= 10; ++i) { /* numero de movimientos de cada pokemon */
+        setTimeout(() => {
+          for (let j = 0; j <= 3; ++j) { /* J = al número de pokemons que quieres que se muevan */
+
+            setTimeout(() => {
+
+              const top = Math.floor(Math.random() * (window.innerHeight - 400 + 1) + 200);
+              const left = Math.floor(Math.random() * (window.innerWidth - 200));
+
+              let sprite = '';
+              let direccion = '';
+
+              if (top >= anteriorY[j] && left >= anteriorX[j]) {
+                sprite = "url(" + DatosService.listaPokemonsVisibles[j].urlSprite + ")";
+                direccion = "scaleX(-1)";
+              }
+              else if (top < anteriorY[j] && left >= anteriorX[j]) {
+                sprite = "url(" + DatosService.listaPokemonsVisibles[j].urlSpriteBack + ")";
+                direccion = "scaleX(1)";
+              }
+              else if (top >= anteriorY[j] && left < anteriorX[j]) {
+                sprite = "url(" + DatosService.listaPokemonsVisibles[j].urlSprite + ")";
+                direccion = "scaleX(1)";
+              }
+              else if (top < anteriorY[j] && left < anteriorX[j]) {
+                sprite = "url(" + DatosService.listaPokemonsVisibles[j].urlSpriteBack + ")";
+                direccion = "scaleX(-1)";
+              }
+
+              anteriorY[j] = top;
+              anteriorX[j] = left;
+
+              this.pokemones[j] = {
+                'margin-left': left + "px",
+                'margin-top': top + "px",
+                'scale': top / window.innerHeight * 3,
+                'background-image': sprite,
+                'transform': direccion,
+              };
+
+              /* this.pokemones[j]['scale'] = 2; pa saber yo a usar mapas */
+
+            }, 500 * j); /* tiempo de un movimiento al del siguiente */
```

Comment:
Y aquí y abajo lo mismo. En general es mejor práctica usar variables con nombres autodescriptivos que meter números sin contexto y un comentario que los expliquen

Reply:
Puedo poner una variable que multiplique el tiempo del bucle del movimiento de cada pokemon por el número de pokemons. De hacer esto me tengo que traer el tiempo de transición del CSS al Ts. Quiero verlo detenidamente para asegurarme que no nos encontramos con otros problemas, pero me parece perfecto. ❤️

Reply:
Simplemente añadir una variable `TIEMPO_ENTRE_MOVIMIENTOS = 500` y luego usarla aquí

```suggestion
            },  this.TIEMPO_ENTRE_MOVIMIENTOS * j);
```